x-pack/plugins/ml/public/embeddables/anomaly_swimlane/anomaly_swimlane_embeddable_factory.tsx

```diff
@@ -0,0 +1,321 @@
+        serialize: serializeTimeRange,
+      } = initializeTimeRange(state);
+
+      const { swimLaneControlsApi, serializeSwimLaneState, swimLaneComparators } =
+        initializeSwimLaneControls(state, titlesApi);
+
+      // Helpers for swim lane data fetching
+      const chartWidth$ = new BehaviorSubject<number | undefined>(undefined);
+
+      const api = buildApi(
+        {
+          ...titlesApi,
+          ...timeRangeApi,
+          ...swimLaneControlsApi,
+          query$,
+          filters$,
+          interval,
+          setInterval: (v) => interval.next(v),
+          dataViews: buildDataViewPublishingApi(
+            {
+              anomalyDetectorService: services[2].anomalyDetectorService,
+              dataViewsService: services[1].data.dataViews,
+            },
+            { jobIds: swimLaneControlsApi.jobIds },
+            subscriptions
+          ),
+          dataLoading,
+          serializeState: () => {
+            return {
+              rawState: {
+                ...serializeTitles(),
+                ...serializeTimeRange(),
+                ...serializeSwimLaneState(),
+              },
+              references: [],
+            };
+          },
+        },
+        {
+          ...timeRangeComparators,
+          ...titleComparators,
+          ...swimLaneComparators,
+        }
+      );
+
+      const appliedTimeRange$: Observable<TimeRange | undefined> = combineLatest([
+        api.timeRange$,
+        apiHasParentApi(api) && apiPublishesTimeRange(api.parentApi)
+          ? api.parentApi.timeRange$
+          : of(null),
+        apiHasParentApi(api) && apiPublishesTimeRange(api.parentApi)
+          ? api.parentApi.timeslice$
+          : of(null),
+      ]).pipe(
+        // @ts-ignore
+        map(([timeRange, parentTimeRange, parentTimeslice]) => {
+          if (timeRange) {
+            return timeRange;
+          }
+          if (parentTimeRange) {
+            return parentTimeRange;
+          }
+          if (parentTimeslice) {
+            return parentTimeRange;
+          }
+          return undefined;
+        })
+      ) as Observable<TimeRange | undefined>;
+
+      const { swimLaneData$, onDestroy } = initializeSwimLaneDataFetcher(
+        api,
+        chartWidth$.asObservable(),
+        dataLoading,
+        blockingError,
+        appliedTimeRange$,
+        query$,
+        filters$,
+        refresh$,
+        anomalySwimLaneServices
+      );
+
+      const onRenderComplete = () => {};
+
+      return {
+        api,
+        Component: () => {
+          const { theme, i18n, uiSettings } = coreStartServices;
+          const { uiActions } = pluginsStartServices;
+
+          const I18nContext = i18n.Context;
+          const timeBuckets = useTimeBuckets(uiSettings);
+
+          useUnmount(() => {
+            onDestroy();
+            subscriptions.unsubscribe();
+          });
+
+          const [fromPage, perPage, swimlaneType, swimlaneData, error] =
```

Comment:
As discussed offline, looks like an issue with the number of rows displayed per page after editing the 'view by' field. For example, if you switch between 'view by region' (4 regions and 4 rows) to 'view by instance' (> 20 instances) for the cloudwatch data set, only 4 instances are initially displayed on page 1:

<img width="1191" alt="Screenshot 2024-04-08 at 17 47 20" src="https://github.com/elastic/kibana/assets/7405507/864b55a4-1424-49ee-95b9-66e1ed823f5b">

Reply:
Fixed in https://github.com/elastic/kibana/pull/179547/commits/fa74d2365cf4aa1256db4f13a779b7109e3f9fa6

Reply:
Looks like you need to reset the page number to 1 if you change the settings. Currently if you are on page 2 of 'view by instance' and then switch to 'view by region' it renders as `No anomalies found`.

Reply:
Confirmed fixed by https://github.com/elastic/kibana/pull/179547/commits/734e98043d3c02af433fdfb98d4e396199cb5c3d